components/script/dom/gpudevice.rs

```diff
@@ -636,6 +650,86 @@ impl GPUDeviceMethods for GPUDevice {
         GPUCommandEncoder::new(&self.global(), self.channel.clone(), encoder, true)
     }
 
+    /// https://gpuweb.github.io/gpuweb/#dom-gpudevice-createtexture
+    fn CreateTexture(&self, descriptor: &GPUTextureDescriptor) -> DomRoot<GPUTexture> {
+        let mut valid = true;
+        let size = match descriptor.size {
+            GPUExtent3D::GPUExtent3DDict(ref dict) => wgt::Extent3d {
+                width: dict.width,
+                height: dict.height,
+                depth: dict.depth,
+            },
+            GPUExtent3D::RangeEnforcedUnsignedLongSequence(ref v) => wgt::Extent3d {
+                width: v[0],
+                height: v[1],
+                depth: v[2],
+            },
+        };
+        let desc = wgt::TextureDescriptor {
+            label: Default::default(),
+            size,
+            mip_level_count: descriptor.mipLevelCount,
+            sample_count: descriptor.sampleCount,
+            dimension: match descriptor.dimension {
+                GPUTextureDimension::_1d => wgt::TextureDimension::D1,
+                GPUTextureDimension::_2d => wgt::TextureDimension::D2,
+                GPUTextureDimension::_3d => wgt::TextureDimension::D3,
+            },
+            format: convert_texture_format(descriptor.format),
+            usage: match wgt::TextureUsage::from_bits(descriptor.usage) {
+                Some(t) => t,
+                None => {
+                    valid = false;
+                    wgt::TextureUsage::empty()
+                },
+            },
+        };
+
+        let texture_id = self
+            .global()
+            .wgpu_id_hub()
+            .lock()
+            .create_texture_id(self.device.0.backend());
+
+        self.channel
+            .0
+            .send(WebGPURequest::CreateTexture {
+                device_id: self.device.0,
+                texture_id,
+                descriptor: desc,
+            })
+            .expect("Failed to create WebGPU Texture");
+
+        let texture = webgpu::WebGPUTexture(texture_id);
+
+        let texture_size = match descriptor.size {
+            GPUExtent3D::GPUExtent3DDict(ref dict) => GPUExtent3DDict {
+                width: dict.width,
```

Comment:
shouldn't this just be `dict.clone()`?

Reply:
`GPUExtent3DDict` does not implement `Clone`. None of the structs generated from webidl dictionary type implement `Clone`.

Reply:
bah, that's sad